superset/commands/dashboard/export.py

```diff
@@ -190,6 +201,5 @@ def _export(
                 if dataset_id is not None:
                     dataset = DatasetDAO.find_by_id(dataset_id)
                     if dataset:
-                        target["datasetUuid"] = str(dataset.uuid)
```

Comment:
At the risk of introducing yet more regressions, I feel it would be cleaner to do the following:
```python
if export_related:
    for target in native_filter...
```

The point being, no sense in iterating over all datasets, if `export_related` is `False`, as we're not doing anything else here (look at how charts are exported above).

Reply:
Done. Since this is now only to invoke the dataset export, it is fine to have the condition before iterating.